instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -415,6 +422,7 @@ func StreamServerInterceptor(opts ...Option) grpc.StreamServerInterceptor {
 		info *grpc.StreamServerInfo,
 		handler grpc.StreamHandler,
 	) error {
+		requestStartTime := time.Now()
```

Comment:
We are duplicating the same logic twice. How about a method setting up a method `recordDuration(time.Time)` which would be executed as a defer:

```
defer recordDuration(time.Now())
```

```
func recordDuration(t time.Time) {
	elapsedTime := float64(time.Since(t)) / float64(time.Millisecond)
	attr = append(attr, semconv.RPCGRPCStatusCodeKey.Int64(int64(status_code)))
	cfg.rpcServerDuration.Record(ctx, elapsedTime, attr...)
}
```

It means status code needs to be set as a method variable so the defer method can access it.